nuxeo-core/nuxeo-core-cache/src/main/java/org/nuxeo/ecm/core/transientstore/api/TransientStoreConfig.java

```diff
@@ -114,4 +121,18 @@ public TransientStore getStore() {
         return store;
     }
 
+    /**
+     * Returns the directory blobs will be stored.
+     *
+     * @since 9.1
+     */
+    public File getDataDir() {
+        if (StringUtils.isBlank(path)) {
+            File transienStoreHome = new File(Environment.getDefault().getData(), "transientstores");
+            return new File(transienStoreHome, getName());
```

Comment:
Don't put any logic in the descriptor, it should be pure data. Do everything in the caller.